### PR TITLE
Epic 7 - Maria Brychko

### PR DESCRIPTION
**Pull request by MariaBrychko**

### CODE REVIEW

**Comment by okohuch** [comment]
стоп

**Comment by izagor** [comment]
+ 12-19-2023

File: ai_12/mariia_brychko/Epic_7/kontrolna.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+const int ROWS = 10;     //оголошення констант
+const int COLS = 10;
+const int NUM_SUMS = 10;
+const int CONSTANT = 1003;
+const std::string FILENAME = "output.txt"; // Константа з іменем файлу
+
+
+// Функція для генерування матриці з цілими числами
+int** generateIntMatrix() {   // Повертає вказівник на двомірний масив цілих чисел - матрицю
```
Comment (fm2lpnu) [inline ai_12/mariia_brychko/Epic_7/kontrolna.cpp:14]: Task 1 (a,b) is not implemented according to requirements.
One of two options is implemented

File: ai_12/mariia_brychko/Epic_7/kontrolna.cpp [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+// Функція для підрахунку сум для випадкових елементів у стовпцях матриці
+// matrix - вказівник на двомірний масив цілих чисел (матрицю), з якою ми працюємо
+// sums - вказівник на одномірний масив цілих чисел, куди зберігаються обчислені суми
+void calculateSums(int** matrix, int* sums) {
+    for (int i = 0; i < COLS; ++i) { // Ітерація через стовпці матриці
+        int sum = 0; // Ініціалізація змінної для зберігання суми
+        for (int j = 0; j < NUM_SUMS; ++j) { // Ітерація для обчислення суми для випадкових елементів
+            int randomRow = rand() % ROWS; // Генерація випадкового рядка
```
Comment (fm2lpnu) [inline ai_12/mariia_brychko/Epic_7/kontrolna.cpp:43]: Mismatch with requirements
3-4 random elements

**Review by fm2lpnu — APPROVED** [review]

**Pull request merged**